controllers/oauthAuthentication.js

```diff
@@ -30,7 +30,15 @@ export const oauthLoginAuthorize = (req, res) => {
 export const oauthLoginCallback = async (req, res, next) => {
   // check if the mandatory Authorization code is there
   if (!req.query.code) {
-    return res.sendStatus(400);
+    /**
+     * @throws the request is not authorized
+     * @see https://www.rfc-editor.org/rfc/rfc6749.html#section-4.1.2.1
+     */
+    const params = {
+      error: 'access_denied',
+      error_description: "vous avez invalidé la tentative d'identification, vous êtes maintenant déconnecté de ce service",
+    };
+    return res.redirect(`/login?${transformParams(params)}`);
```

Comment:
this redirection imply amount of code that can be avoided. Can we just render an error page here ?

```
return res.render('error.ejs').status(400)
```

with pages/error.ejs being like:
```
<section class="hero notification is-danger has-text-centered">
  ...
  <%= errorDescription %>
  <a href="/">Retour à l'acceuil</a>
  ...
</section>
```

We must keep in mind that this project intended to be simpler as possible so non javascript developer can understand it.

Reply:
https://www.rfc-editor.org/rfc/rfc6749.html#section-4.1.2.1
If you want to change the OpenID process.... good luck :)